components/ledger/ledger-utxo-flow/src/test/kotlin/net/corda/ledger/utxo/flow/impl/flows/transactiontransmission/SendAsLedgerTransactionFlowTest.kt

```diff
@@ -0,0 +1,108 @@
+import net.corda.v5.base.exceptions.CordaRuntimeException
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+class SendAsLedgerTransactionFlowTest {
+
+    private companion object {
+        val TX_ID = SecureHashImpl("SHA", byteArrayOf(1, 1, 1, 1))
+    }
+
+    private val flowEngine = mock<FlowEngine>()
+    private val flowMessaging = mock<FlowMessaging>()
+
+    private val sessionAlice = mock<FlowSession>()
+    private val sessionBob = mock<FlowSession>()
+    private val sessions = listOf(sessionAlice, sessionBob)
+
+    private val transaction = mock<UtxoSignedTransactionInternal>()
+    private val successMessage = "Successfully received transaction."
+
+    @BeforeEach
+    fun beforeEach() {
+        whenever(transaction.id).thenReturn(TX_ID)
+        whenever(flowEngine.subFlow(any<TransactionBackchainSenderFlow>())).thenReturn(Unit)
+    }
+
+    @Test
+    fun `does nothing when receiving payload successfully`() {
+        whenever(sessionAlice.receive(Payload::class.java)).thenReturn(
+            Payload.Success(successMessage)
+        )
+        whenever(sessionBob.receive(Payload::class.java)).thenReturn(
+            Payload.Success(successMessage)
+        )
+
+        callSendTransactionFlow(transaction, sessions)
+
+        verify(flowMessaging).sendAll(transaction.wireTransaction, sessions.toSet())
+        verify(sessionAlice).receive(Payload::class.java)
+    }
+
+    @Test
+    fun `sending transaction with dependencies should call backchain flow`() {
+        whenever(transaction.inputStateRefs).thenReturn(listOf(mock()))
+
+        whenever(sessionAlice.receive(Payload::class.java)).thenReturn(
+            Payload.Success(successMessage)
+        )
+        whenever(sessionBob.receive(Payload::class.java)).thenReturn(
+            Payload.Success(successMessage)
+        )
+
+        callSendTransactionFlow(transaction, sessions)
+
+        verify(flowEngine).subFlow(TransactionBackchainSenderFlow(TX_ID, sessionAlice))
+    }
+
+    @Test
+    fun `sending transaction with no dependencies should not call backchain flow`() {
+        whenever(sessionAlice.receive(Payload::class.java)).thenReturn(
+            Payload.Success(successMessage)
+        )
+        whenever(sessionBob.receive(Payload::class.java)).thenReturn(
+            Payload.Success(successMessage)
+        )
+
+        callSendTransactionFlow(transaction, sessions)
+
+        verify(flowEngine, never()).subFlow(TransactionBackchainSenderFlow(TX_ID, sessionAlice))
+    }
+
+    @Test
+    fun `sending unverified transaction should throw exception`() {
```

Comment:
I think
```suggestion
    fun `exceptions get propagated back from receiver`() {
```
would be a bit more correct.
(Sending unverified transactions is OK with these new functions.)